rust/src/main.rs

```diff
@@ -11,19 +11,48 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-use std::io;
 
 #[macro_use]
 extern crate log;
 extern crate chrono;
 extern crate fern;
 
+extern crate dirs;
+
 extern crate xi_core_lib;
 extern crate xi_rpc;
 
+use std::io;
+use std::fs;
+use std::path::PathBuf;
+
 use xi_core_lib::XiCore;
 use xi_rpc::RpcLoop;
 
+const XI_DIRECTORY: &'static str = "xi-core";
```

Comment:
I believe that clippy will warn that the `&'static` is redundant here. You used to need it, but this changed sometime this year.

Reply:
I still need the `&str` as `const` must be annotated with a type but the `'static` lifetime is indeed redundant.
I've ran the fixed version by clippy and didn't have any complaints this time.
`const XI_DIRECTORY: &str = "xi-core";`